xstac/_xstac.py

```diff
@@ -3,18 +3,38 @@
 """
 import copy
 import json
+import logging
 from pystac import stac_object
 import xarray as xr
 import numpy as np
 import pystac
 import pandas as pd
 from pyproj import CRS, Transformer
-from typing import List, Dict
+from typing import Dict
 
 from ._types import TemporalDimension, HorizontalSpatialDimension, Datacube, Variable
 
+logger = logging.getLogger(__name__)
+
 SCHEMA_URI = "https://stac-extensions.github.io/datacube/v2.0.0/schema.json"
 
+CF_STANDARD_AXES = dict(temporal_dimension="T", x_dimension="X", y_dimension="Y")
+
+
+def maybe_use_cf_standard_axis(kw, kw_name, ds):
+    if kw is None:
+        try:
+            kw = ds.cf[CF_STANDARD_AXES[kw_name]].name
+        except KeyError:
```

Comment:
I'm not sure how to handle this case. In general, I'd assume people don't look at logs and so just logging the warning isn't great, and might just annoy people who don't want that dimension.

How about this: the default of `None` will mean get it from the CF conventions by default, but we just raise this KeyError if it's not present for `time`, `y`, and `x`.

Users who have those dims, but don't / can't use CF conventions can specify `time="my-time-dim"`.

Users who *don't* have those dims can drop them explicitly with `time=False`.

So I think the whole `try / except` can be removed, or you could use it to reraise with a better error message

```python
try:
    ...
except KeyError as e:
    raise KeyError("better message about setting {dim}=False to drop") from e
```

Reply:
Added two descriptive errors in https://github.com/TomAugspurger/xstac/pull/20/commits/c3e1001a9253dddb7a2ae0113ff278550e21eac6